src/App.tsx

```diff
@@ -1,61 +1,115 @@
+import React, { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+const getUserById = (userId: number): User | null => (
+  usersFromServer.find(user => user.id === userId) || null
+);
+
+const todosWithUser = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+const getNewId = (array: { id: number }[]) => (
+  Math.max(...array.map(todo => todo.id)) + 1
+);
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosWithUser);
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [enteredTitle, setEnteredTitle] = useState('');
+  const [selectUserError, setSelectUserError] = useState(false);
+  const [enteredTitleError, setEnteredTitleError] = useState(false);
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUserId(+event.currentTarget.value);
+    setSelectUserError(false);
```

Comment:
```suggestion
   if (selectUserError) {
      setSelectUserError(false);
    }
```

I guess we need to set error false only if it is true